Fix unix Makefile template to avoid command line too long error

sunnyqeen opened the pull request:
On cygwin/msys system that run on Windows, command line length is limited. using response file instead of putting objects on the command line will avoid this error.
I got this error when building for android platform on msys/cygwin

levitte commented:
Unfortunately, this makes assumptions on the `make` implementation as well as on the compiler, which aren't as universal as you might think.

`$(file ...)` is a GNU make thing.

`@file` is supported by gcc and clang (or actually `ld` in this case, gcc and clang are just passing on the argument), but unsure about other C compilers or linkers...

Note that I don't mind a change of this sort, but it will need to be done in a way that takes toolchain varieties into account.

levitte commented:
That would be suitable.  You have a function at the start of unix-Makefile.tmpl, named `windowsdll`, which may be useful.  Not the greatest name for your purpose, but it should serve you well enough.

sunnyqeen commented:
added run_on_windows check

tom-cosgrove-arm commented on Configurations/unix-Makefile.tmpl:
```diff
@@ -1854,13 +1855,27 @@ EOF
 EOF
       }
+      if (!run_on_windows()) {
       $recipe .= <<"EOF";
```
Indentation seems wrong now that an `if()` has been added on the line above - just this line and (new) line 1867 should have further indents

sunnyqeen replied:
Fixed

sunnyqeen committed:
Fix unix Makefile template to avoid command line too long error on windows
On cygwin/msys systems that run on Windows, command line length is limited. using response file instead of putting objects on the command line will avoid this error

CLA: trivial

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

openssl-machine commented:
This pull request is ready to merge

nhorman commented:
merged to master, thank you
